core/transcoder.go

```diff
@@ -16,6 +19,78 @@ import (
 	"github.com/golang/glog"
 )
 
+var testSegment = []byte{0x1f, 0x8b, 0x8, 0x8, 0xfc, 0xdd, 0x90, 0x5e, 0x0, 0x3, 0x7a, 0x7a, 0x7a, 0x7a, 0x2e, 0x6d,
```

Comment:
It might be good to put this in a separate file with a reproducible way to generate the segment. I was able to get a much smaller (182 byte) segment like this:

```sh
#!/usr/bin/env bash

FILE=testSegment.go

echo "package core" > $FILE
echo "// Auto generated by $0 DO NOT EDIT" >> $FILE
echo "var testSegment = []byte{" >> $FILE
ffmpeg -loglevel quiet -f lavfi -i color=white:s=16x16 -vframes 1 \
    -metadata title='!' -metadata service_provider='!' -f mpegts - \
    | gzip | xxd -i | head -c -1 >> $FILE
echo "}" >> $FILE
```

We can check in both the script and the resulting `testSegment.go` and regenerate it only when needed. (So most users would not need any of these utilities installed unless they're actually regenerating the file. They are fairly standard in most distros anyway except for ffmpeg.)

Reply:
But I thought we still can't transcode one-frame segments? That's why I've made segment with 12 frames - just to be sure it will transcode successfully.

Reply:
https://github.com/livepeer/lpms/issues/168 is being worked on by @jailuthra so that will be resolved sooner rather than later.

Do we actually need to complete a transcode though? Wouldn't simply attempting to open the GPU be enough to trigger a failure? This would detect both invalid devices and hitting the session cap; these cases wouldn't actually go all the way to a complete transcode.

Reply:
> Wouldn't simply attempting to open the GPU be enough to trigger a failure? 

Do we have LPMS API to just try 'open the GPU'? 
But anyway, I think it is good to be sure that we really can transcode, not just open GPU.